R/content-type.R

```diff
@@ -0,0 +1,115 @@
+#' Check the content type of a response
+#'
+#' A different content type than expected often leads to an error in parsing
+#' the response body. This function checks that the content type of the response
+#' is as expected and fails otherwise.
+#'
+#' @param resp A response object.
+#' @param valid_types A character vector of valid content types.
+#' @param valid_suffix A string given an "structured media type" suffix.
+#' @param check_type Should the type actually be checked? Provided as a
+#'   convenience for when using this function inside `resp_body_*` helpers.
+#' @inheritParams rlang::args_error_context
+#' @return Called for its side-effect; erroring if the response does not
+#'   have the expected content type.
+#' @export
+#' @examples
+#' resp <- response(headers = list(`content-type` = "application/json"))
+#' check_resp_content_type(resp, "application/json")
+#' try(check_resp_content_type(resp, "application/xml"))
+#'
+#' # `types` can also specify multiple valid types
+#' check_resp_content_type(resp, c("application/xml", "application/json"))
+check_resp_content_type <- function(resp,
```

Comment:
I wonder if this should be `resp_check_content_type()`; the `resp`/`req` prefix feels very strong in httr2.